client/app/services/query.js

```diff
@@ -43,6 +43,40 @@ class QueryResultError {
 }
 
 
+class Parameter {
+  constructor(parameter) {
+    this.title = parameter.title;
+    this.name = parameter.name;
+    this.type = parameter.type;
+    this.value = parameter.value;
+    this.global = parameter.global;
+  }
+
+  get ngModel() {
+    if (this.type === 'date' || this.type === 'datetime-local' || this.type === 'datetime-with-seconds') {
+      this.$$value = this.$$value || moment(this.value).toDate();
+      return this.$$value;
+    } else if (this.type === 'number') {
+      this.$$value = this.$$value || parseInt(this.value, 10);
+      return this.$$value;
+    }
+
+    return this.value;
+  }
+
+  set ngModel(value) {
+    if (this.type === 'date') {
+      this.value = moment(value).format('YYYY-MM-DD');
+      this.$$value = moment(this.value).toDate();
+    } else if (this.type === 'datetime-local' || this.type === 'datetime-with-seconds') {
```

Comment:
Why is there two versions of this if they are handled the same? If a deep link has seconds in the query parameters,then the datetime-local type will show seconds in the datepicker.